kapitan/cli.py

```diff
@@ -361,14 +364,33 @@ def secret_update(args, ref_controller):
                 logger.warning("WARNING: parameters.kapitan.secrets.recipients is deprecated, " +
                     "please use parameters.kapitan.secrets.gpg.recipients")
                 recipients = inv['nodes'][args.target_name]['parameters']['kapitan']['secrets']['recipients']
+        if not recipients:
+            raise KapitanError("No GPG recipients specified. Use --recipients or specify in " +
+                               "parameters.kapitan.secrets.gpg.recipients and use --target")
         type_name, token_path = token_name.split(":")
         tag = '?{{gpg:{}}}'.format(token_path)
         secret_obj = ref_controller[tag]
         secret_obj.update_recipients(recipients)
         ref_controller[tag] = secret_obj
-    # TODO: Implement --update for KMS
+
+    elif token_name.startswith("gkms:"):
+        key = args.key
+        if args.target_name:
+            inv = inventory_reclass(args.inventory_path)
+            if 'secrets' not in inv['nodes'][args.target_name]['parameters']['kapitan']:
+                raise KapitanError("parameters.kapitan.secrets not defined in {}".format(args.target_name))
+
+            key = inv['nodes'][args.target_name]['parameters']['kapitan']['secrets']['gkms']['key']
+        if not key:
+            raise KapitanError("No KMS key specified. Use --key or specify in parameters.kapitan.secrets.gkms.key and use --target")
```

Comment:
`Use --key or specify *it* in parameters.kapitan.secrets.gkms.key`

Reply:
You can move it to line 378. It's easier to read and you can fail faster, before doing any computation.